Upstream: unknown
Source: Gentoo
Reason: verification pass showed wrong file count

http://bugs.gentoo.org/99845
original patch by Robin H. Johnson (fixed formatting)

--- dosfsck/dosfsck.c.orig	2004-02-25 20:28:57.000000000 +0100
+++ dosfsck/dosfsck.c	2008-09-08 19:42:27.000000000 +0200
@@ -85,8 +85,9 @@
     DOS_FS fs;
     int rw,salvage_files,verify,c;
     unsigned long free_clusters;
-    
-    rw = salvage_files = verify = 0;
+    unsigned n_files_check, n_files_verify;
+
+    rw = salvage_files = verify = n_files_check = n_files_verify = 0;
     interactive = 1;
     check_atari();
 
@@ -154,12 +155,15 @@
     free_clusters = update_free(&fs);
     file_unused();
     qfree(&mem_queue);
+    n_files_check = n_files;
     if (verify) {
+	n_files = 0;
 	printf("Starting verification pass.\n");
 	read_fat(&fs);
 	scan_root(&fs);
 	reclaim_free(&fs);
 	qfree(&mem_queue);
+	n_files_verify = n_files;
     }
 
     if (fs_changed()) {
